README.md

```diff
@@ -1,162 +1,162 @@
+<img src="assets/wide-logo.png">
 
-With Shards we aim to create a low-code tool for producing high-performance, multi-platform applications. Our approach is to create a scripting tool that is both visual and textual. This tool will represent the flow of data and logic as it is, in a "what you see is how it works" manner.
+Shards is a dataflow visual programming language that makes app development accessible to all. It uses a straightforward system of connecting visual blocks, called shards, to build full-fledged apps and games without coding. Wires powered by a lightweight cooperative concurrency model, glue sequences of shards together, preventing the complicated spider webs usually associated with visual development. Automatic type checking provides reliability, while optimized shard implementations ensure high performance. The intuitive visual workflow allows beginners to quickly build powerful, multi-platform apps.
 
-## Introduction
+In Shards, the code's syntax and its computational graph of interconnected shards and wires semantically represent the same visual model. This synergy allows for seamless round-trip engineering, with visual development applications capable of visually loading the computational graph from the code and subsequently generating its code representation.
 
-Shards is a concatenative programming language that emphasizes modularity and composition. This is achieved through the use of shards, which represent individual functional units that can easily be connected with other shards to form complex computational workflows. Each shard can have parameters, which function like properties that manipulate how the shard itself works. This means that developers can create a diverse range of workflows that can be easily customized to meet specific requirements, enabling faster and more efficient deployment of applications.
+Shards powers an upcoming AI-powered game creation system where communities can play and collaborate in real time. While Shards is often not explicitly seen there, it is the language behind the AI-assisted visual interactions that will allow users to create games and experiences in a low to no-code environment.
 
-Shards shares some similarities with stack-based languages such as Forth, as both focus on functional composition and modularity. However, Shards has a unique way of executing code that sets it apart from other languages.
+In Shards, every primitive is a shard, flowing from shard to shard, to build a computational graph that represents a visual model. Example:
 
-By using Shards, developers can create a diverse range of computational workflows that can be easily customized to meet specific requirements, enabling faster and more efficient deployment of applications.
+```dart
+[[1 2 3] [2 3 4] [3 4 5]] | Reduce(Math.Add($0)) | Assert.Is([6 9 12]) | Log
```

Comment:
Why does it say `$0`? shouldn't it just be a number?

Reply:
ah yeah that's correct but not sure this sample is ideal, the `$0` is a special variable in this case for Reduce, injected.